lib/virtus/attribute/nullify_blank.rb

```diff
@@ -0,0 +1,24 @@
+module Virtus
+  class Attribute
+
+    # Attribute extension which nullifies blank attributes when coercion failed
+    #
+    module NullifyBlank
+
+      # @see [Attribute#coerce]
+      #
+      # @api public
+      def coerce(input)
+        output = super
+
+        if !value_coerced?(output) && input.to_s.empty?
+          nil
+        else
+          output
+        end
+      end
```

Comment:
This could cover most scenarios but I wonder if when the attribute is a String we really want a `nil` instead of the empty value.